lib/tracing-limit/src/lib.rs

```diff
@@ -178,39 +213,125 @@ where
 
                 events.insert(id, state);
             }
+
+            self.inner.on_event(event, ctx);
+        }
+    }
+
+    #[inline]
+    fn on_enter(&self, id: &span::Id, ctx: Context<'_, S>) {
+        self.inner.on_enter(id, ctx);
+    }
+
+    #[inline]
+    fn on_exit(&self, id: &span::Id, ctx: Context<'_, S>) {
+        self.inner.on_exit(id, ctx);
+    }
+
+    #[inline]
+    fn on_close(&self, id: span::Id, ctx: Context<'_, S>) {
+        self.inner.on_close(id, ctx);
+    }
+
+    #[inline]
+    fn on_id_change(&self, old: &span::Id, new: &span::Id, ctx: Context<'_, S>) {
+        self.inner.on_id_change(old, new, ctx);
+    }
+}
+
+impl<S, L> RateLimitedLayer<S, L>
+where
+    S: Subscriber,
+    L: Layer<S>,
+{
+    fn create_event(
+        &self,
+        ctx: &Context<S>,
+        metadata: &'static Metadata<'static>,
+        message: String,
+        rate_limit: u64,
+    ) {
+        let fields = metadata.fields();
+
+        let message = display(message);
+
+        if let Some(message_field) = fields.field("message") {
+            let values = [(&message_field, Some(&message as &dyn Value))];
+
+            let valueset = fields.value_set(&values);
+            let event = Event::new(metadata, &valueset);
+            self.inner.on_event(&event, ctx.clone());
+        } else {
+            let values = [(
+                &fields.field(RATE_LIMIT_SECS_FIELD).unwrap(),
+                Some(&rate_limit as &dyn Value),
+            )];
+
+            let valueset = fields.value_set(&values);
+            let event = Event::new(metadata, &valueset);
+            self.inner.on_event(&event, ctx.clone());
         }
     }
 }
 
+#[derive(Debug)]
+struct State {
+    start: Option<Instant>,
```

Comment:
This `Option` isn't necessary anymore because it's always `Some` now.

Reply:
Indeed, thank you 😄